verification/cbmc/include/core_json_annex.h

```diff
@@ -67,45 +67,71 @@
-                 size_t max );
-
-bool skipSpaceAndComma( const char * buf,
-                        size_t * start,
-                        size_t max );
+JSONStatus_t skipCollection( const char * buf,
+                             size_t * start,
+                             size_t max )
+__CPROVER_requires( max > 0 && max < CBMC_MAX_BUFSIZE )
+__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
+__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
+__CPROVER_assigns( *start )
+__CPROVER_ensures( skipCollectionEnum( __CPROVER_return_value ) )
+__CPROVER_ensures( __CPROVER_return_value == JSONSuccess ==> *start <= max );
+
+void skipScalars( const char * buf,
+                  size_t * start,
+                  size_t max,
+                  char mode )
+__CPROVER_requires( max > 0 && max < CBMC_MAX_BUFSIZE )
+__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
+__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
+__CPROVER_requires( ( mode == '{' ) || ( mode == '[' ) )
+/* __CPROVER_requires(isOpenBracket_( mode )) */
+__CPROVER_assigns( *start )
+__CPROVER_ensures( *start >= __CPROVER_old( *start ) )
+/* __CPROVER_ensures(__CPROVER_old(*start) <= max ==> *start <= max) */
+__CPROVER_ensures( __CPROVER_old( *start ) > max ==> *start == __CPROVER_old( *start ) );
 
 bool skipAnyScalar( const char * buf,
                     size_t * start,
-                    size_t max );
+                    size_t max )
+__CPROVER_requires( max > 0 && max < CBMC_MAX_BUFSIZE )
+__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
+__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
+__CPROVER_assigns( *start )
+__CPROVER_ensures( isBool( __CPROVER_return_value ) )
+__CPROVER_ensures( *start >= __CPROVER_old( *start ) )
+__CPROVER_ensures( __CPROVER_old( *start ) <= max ==> *start <= max )
+__CPROVER_ensures( __CPROVER_old( *start ) > max ==> *start == __CPROVER_old( *start ) );
 
-JSONStatus_t skipCollection( const char * buf,
-                             size_t * start,
-                             size_t max );
+void skipSpace( const char * buf,
+                size_t * start,
+                size_t max )
+__CPROVER_requires( max > 0 && max < CBMC_MAX_BUFSIZE )
+__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
+__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
+__CPROVER_assigns( *start )
+__CPROVER_ensures( *start >= __CPROVER_old( *start ) )
+__CPROVER_ensures( __CPROVER_old( *start ) <= max ==> *start <= max )
+__CPROVER_ensures( __CPROVER_old( *start ) > max ==> *start == __CPROVER_old( *start ) );
+
+bool skipString( const char * buf,
+                 size_t * start,
+                 size_t max )
+__CPROVER_requires( max > 0 && max < CBMC_MAX_BUFSIZE )
+__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
+__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
+__CPROVER_assigns( *start )
+__CPROVER_ensures( isBool( __CPROVER_return_value ) )
```

Comment:
I wish "jsonValidateEnum" was called "isValidJsonEnum" for the same reason I prefer "isBool" to "validateBool".

Reply:
I've implemented as suggested (across all defines).